internal/cmd/controller/helmops/reconciler/helmop_controller_test.go

```diff
@@ -358,75 +366,167 @@ func TestReconcile_Validate(t *testing.T) {
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "helmop",
+				Namespace: "default",
+			},
+			Spec: fleet.HelmOpSpec{
+				BundleSpec: fleet.BundleSpec{
+					BundleDeploymentOptions: fleet.BundleDeploymentOptions{
+						Helm: &fleet.HelmOptions{
+							Repo:    svr.URL,
+							Chart:   "alpine",
+							Version: "0.1.0", // static version
+						},
+					},
+				},
+				InsecureSkipTLSverify: true,
+			},
+		}
 
-	res, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: namespacedName})
-	if err == nil {
-		t.Errorf("expecting error, got nil")
-	}
-	if err.Error() != "this is a test error" {
-		t.Errorf("expecting error: [this is a test error], got %v", err.Error())
-	}
-	if res.RequeueAfter != 0 {
-		t.Errorf("expecting no requeue when there's an error, but got RequeueAfter: %v", res.RequeueAfter)
-	}
+		namespacedName := types.NamespacedName{Name: helmop.Name, Namespace: helmop.Namespace}
+		client := mocks.NewMockK8sClient(mockCtrl)
+		client.EXPECT().Get(gomock.Any(), gomock.Any(), OfType(&fleet.HelmOp{}), gomock.Any()).Times(1).DoAndReturn(
+			func(ctx context.Context, req types.NamespacedName, fh *fleet.HelmOp, opts ...interface{}) error {
+				fh.Name = helmop.Name
+				fh.Namespace = helmop.Namespace
+				fh.Spec = helmop.Spec
+				fh.Status = fleet.HelmOpStatus{}
+
+				controllerutil.AddFinalizer(fh, finalize.HelmOpFinalizer)
+				return nil
+			},
+		)
+
+		client.EXPECT().Get(gomock.Any(), gomock.Any(), OfType(&fleet.Bundle{}), gomock.Any()).AnyTimes().DoAndReturn(
+			func(ctx context.Context, req types.NamespacedName, bundle *fleet.Bundle, opts ...interface{}) error {
+				bundle.Spec.HelmOpOptions = &fleet.BundleHelmOptions{}
+				return nil
+			},
+		)
+
+		client.EXPECT().Get(gomock.Any(), gomock.Any(), OfType(&fleet.HelmOp{}), gomock.Any()).AnyTimes().DoAndReturn(
+			func(ctx context.Context, req types.NamespacedName, helmOp *fleet.HelmOp, opts ...interface{}) error {
+				return nil
+			},
+		)
+
+		statusClient := mocks.NewMockSubResourceWriter(mockCtrl)
+		client.EXPECT().Status().Return(statusClient).Times(1)
+		statusClient.EXPECT().Patch(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Do(
+			func(ctx context.Context, helmop *fleet.HelmOp, patch crclient.Patch, opts ...interface{}) {
+				c, found := getCondition(helmop, fleet.HelmOpAcceptedCondition)
+				if !found {
+					t.Errorf("expecting to find the %s condition and could not find it.", fleet.HelmOpAcceptedCondition)
+				}
+				if !strings.Contains(c.Message, "404") {
+					t.Errorf("expecting message to contain [404] in condition, got [%s]", c.Message)
+				}
+			},
+		).Times(1)
+
+		r := HelmOpReconciler{
+			Client: client,
+			Scheme: scheme,
+		}
+
+		ctx := context.TODO()
+
+		res, err := r.Reconcile(ctx, ctrl.Request{NamespacedName: namespacedName})
+		if err == nil {
+			t.Errorf("expecting error, got nil")
+		}
+		if !strings.Contains(err.Error(), "404") {
+			t.Errorf("expecting error containing: [404], got %v", err.Error())
+		}
```

Comment:
nit: Maybe it would be better to check the full error message here.
I mention this because if the error body is still being returned, this check would pass.
If we check for the full string (as done in the unit tests above), it wouldn't even be necessary to verify that the HTML is not present.